desc/protoprint/print.go

```diff
@@ -301,6 +304,12 @@ func (p *Printer) printProto(dsc desc.Descriptor, out io.Writer) error {
 	er := dynamic.ExtensionRegistry{}
 	er.AddExtensionsFromFileRecursively(dsc.GetFile())
 	mf := dynamic.NewMessageFactoryWithExtensionRegistry(&er)
+
+	if r, err := createTypeRegistry(dsc.GetFile()); err == nil {
+		p.registry = r
```

Comment:
So I added the resolver to the printer struct so that I would have access to it during `p.printOption`. This was just to get it working but I do think needs to be removed. Should I just pipe the var all the way through?

Reply:
The printer struct is just configuration and is not stateful. It should instead be woven through the call stack, in lieu of the message factory.